config-service-impl/src/test/java/org/hypertrace/config/service/store/DocumentConfigStoreTest.java

```diff
@@ -368,41 +369,75 @@ void writeAllConfigs() throws IOException {
+    // Create a mock for CloseableIterator<Document>
+    CloseableIterator<Document> mockIterator = mock(CloseableIterator.class);
+    when(mockIterator.hasNext()).thenReturn(false); // No documents to process
 
-    // Verify default sorting is applied
-    assertEquals(2, query.getSorts().size());
-    assertEquals(
-        "configVersion",
-        ((IdentifierExpression) query.getSorts().get(0).getExpression()).getName());
-    assertEquals(
-        "creationTimestamp",
-        ((IdentifierExpression) query.getSorts().get(1).getExpression()).getName());
+    // Mock the collection.query to return the mock iterator
+    when(collection.query(any(Query.class), any(QueryOptions.class))).thenReturn(mockIterator);
+
+    // Act
+    configStore.getAllConfigs(configResource, filter, pagination, Collections.emptyList());
+
+    // Capture the Query object passed to collection.query
+    ArgumentCaptor<Query> queryCaptor = ArgumentCaptor.forClass(Query.class);
+    verify(collection).query(queryCaptor.capture(), any(QueryOptions.class));
+
+    // Assert the properties of the captured Query
+    Query capturedQuery = queryCaptor.getValue();
+    assertNotNull(capturedQuery);
+
+    List<SortingSpec> expectedSorts =
+        Arrays.asList(
+            SortingSpec.of(
+                IdentifierExpression.of("configVersion"),
+                org.hypertrace.core.documentstore.expression.operators.SortOrder.DESC),
+            SortingSpec.of(
+                IdentifierExpression.of("creationTimestamp"),
+                org.hypertrace.core.documentstore.expression.operators.SortOrder.DESC));
+    assertEquals(expectedSorts, capturedQuery.getSorts());
   }
 
   @Test
-  void buildQuery_withCustomPagination() {
+  void buildQuery_withCustomPagination() throws IOException {
     ConfigResource configResource =
         new ConfigResource(RESOURCE_NAMESPACE, RESOURCE_NAME, "tenant1");
     Filter filter = Filter.getDefaultInstance();
     Pagination pagination = Pagination.newBuilder().setOffset(10).setLimit(20).build();
     List<org.hypertrace.config.service.v1.SortBy> sortByList = Collections.emptyList();
 
-    Query query = configStore.buildQuery(configResource, filter, pagination, sortByList);
+    // Create a mock for CloseableIterator<Document>
+    CloseableIterator<Document> mockIterator = mock(CloseableIterator.class);
+    when(mockIterator.hasNext()).thenReturn(false); // No documents to process
+
+    // Mock the collection.query to return the mock iterator
+    when(collection.query(any(Query.class), any(QueryOptions.class))).thenReturn(mockIterator);
 
-    assertEquals(10, query.getPagination().get().getOffset());
-    assertEquals(20, query.getPagination().get().getLimit());
+    // Act
+    configStore.getAllConfigs(configResource, filter, pagination, sortByList);
+
+    // Capture the Query object passed to collection.query
+    ArgumentCaptor<Query> queryCaptor = ArgumentCaptor.forClass(Query.class);
```

Comment:
arg captors aren't needed here (they're almost always overkill). Let me post a branch to illustrate what I was suggesting

Reply:
https://github.com/hypertrace/config-service/pull/290

If that looks good to you, feel free to merge it into your branch (or just take bits and pieces)

Reply:
I have merged your PR. Thanks for that.